modules/seqexec/server/src/main/scala/seqexec/server/package.scala

```diff
@@ -178,8 +179,11 @@ package object server {
     def apply[F[_]: Sync, A](a: => A): SeqActionF[F, A] =
       EitherT(Sync[F].delay(TrySeq(a)))
     def liftF[F[_]: Functor, A](a:          => F[A]): SeqActionF[F, A] = EitherT.liftF(a)
-    def liftIO[F[_]: LiftIO: Functor, A](a: => IO[A]): SeqActionF[F, A] =
-      EitherT.liftF(LiftIO[F].liftIO(a))
+    def liftIO[F[_]: LiftIO, A](a: => IO[A]): SeqActionF[F, A] =
+      EitherT(LiftIO[F].liftIO(a.attempt.map(_.leftMap {
+        case e: SeqexecFailure => e
+        case r                 => SeqexecFailure.SeqexecException(r)
+      })))
```

Comment:
I would call this something else because we already have `LiftIO[EitherT[IO, ...` that has a different behavior. Maybe `embed` or something, to say we're absorbing this `IO` action but we're turning it into a `SeqAction` with different exception-handling behavior. Kind of hope this goes away when we get rid of all the stuff in `EitherT`.

You might want to `flatMap(_.leftMap` and only absorb `NonFatal(r)` and then have a catch-all for fatal exceptions that re-raises them.